src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,249 @@
+     * @return welcome message.
+     */
+    public static String showWelcome() {
+        return "Hello! I'm Duke\nWhat can I do for you?";
+    }
+
+    /**
+     * Gives the program the next line which was input by the user.
+     *
+     * @return string of the input.
+     */
+    public String readCommand() {
+        return input.nextLine();
+    }
+
+
+    /**
+     * Displays the tasks currently in the task list.
+     *
+     * @param taskList the current task list.
+     * @return the tasks in the list.
+     */
+    public static String listTasks(ArrayList<Task> taskList) {
+        if (taskList.size() == 0) {
+            return "You have no tasks in your list!\nTry adding some tasks with todo, event, or deadline.";
+        } else {
+            String output = "Here are the tasks in your list:\n";
+            for (int i = 0; i < taskList.size(); i++) {
+                output += (i + 1) + "." + taskList.get(i) + "\n";
+            }
+            return output;
+        }
+    }
+
+
+    /**
+     * Returns the String message instead of printing
+     *
+     * @param task task to be completed
+     * @return String of the message
+     */
+    public static String finishTask(Task task) {
+        String str = "Nice! I've marked this task as done:\n";
+        str = str + task;
+        return str;
+    }
+
+    /**
+     * Displays the remaining tasks in the list.
+     *
+     * @param tasks task list.
+     * @return the remaining tasks in the list.
+     */
+    public static String remainingTasks(ArrayList<Task> tasks) {
+        return String.format("Now you have %d tasks in the list.", tasks.size());
+    }
+
+    /**
+     * Displays a message when a task is deleted.
+     *
+     * @param task task to be deleted.
+     * @return message when task is deleted.
+     */
+    public static String deleteTask(Task task) {
+        String out = "Noted. I've removed this task:\n";
+        out += task;
+        return out;
+    }
+
+    /**
+     * Displays the exit message for Duke.
+     *
+     * @return exit message.
+     */
+    public static String endDuke() {
+        return "Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Displays the error message for a number format exception.
+     *
+     * @return number format exception message.
+     */
+    public static String numberFormatExceptionMessage() {
+        return "☹ OOPS!!! The value you inputted is not valid!";
+    }
+
+    /**
+     * Displays the error message for a date time parse exception.
+     *
+     * @return date time parse exception message.
+     */
+    public static String dateTimeParseExceptionMessage() {
+        return "☹ OOPS!!!  You used an invalid date! Hint: Use 'YYYY-MM-DD HH:mm'";
+    }
+
+    /**
+     * Displays the error message for an array index out of bounds exception.
+     *
+     * @return array out of bounds message.
+     */
+    public static String arrayIndexOutOfBoundsExceptionMessage() {
```

Comment:
Perhaps another possible way to achieve the same functionalities as these multiple methods might be to declare these strings as class constants and then refer to them? (That might reduce the number of methods listed here!)